perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -2857,4 +2787,77 @@ private boolean checkGroupsCycle(PerunSession sess, int resultGroupId, int opera
 
 		return false;
 	}
+
+	/**
+	 * Method sets subGroups names by their parent group
+	 * Private method for moving groups
+	 *
+	 * @param sess perun session
+	 * @param subGroups groups with same parent group
+	 * @throws InternalErrorException
+	 */
+	private void setSubGroupsNames(PerunSession sess, List<Group> subGroups) throws InternalErrorException {
+		for (Group subGroup : subGroups) {
+			try {
+				subGroup.setName(getParentGroup(sess, subGroup).getName() + ":" + subGroup.getShortName());
+			} catch (ParentGroupNotExistsException ex) {
+				//that should never happen
+				throw new InternalErrorException("Parent group does not exists for group " + subGroup);
+			}
+			//we have to update each subGroup name in database
+			getGroupsManagerImpl().updateGroupName(sess, subGroup);
+
+			// create auditer message for every updated group
+			getPerunBl().getAuditer().log(sess, "{} updated.", subGroup);
+
+			setSubGroupsNames(sess, getSubGroups(sess, subGroup));
+		}
+	}
+
+	/**
+	 * Method remove old relation between moving group and its parent,
+	 * also create new relation between destination and moving group if destination is not null
+	 * Private method for moving groups
+	 *
```

Comment:
Maybe add note to the javadoc, that after calling this method, proper name and parentGroupId must be set to the moving group for keeping DB consistency !